### PR TITLE
Validator#regular expression

### PR DESCRIPTION
**Pull request by Flixow**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +38,8 @@ const rules = {
   username: 'required|unique:users,username',
   email: 'required|email',
   age: 'required|numeric|min:18',
-  gender: 'in:male,female'
+  gender: 'in:male,female',
+  phoneNumber: 'regex:[0-9]{3} ?-?[0-9]{3} ?-?[0-9]{3}'
```
Comment (Idered) [inline README.md]: Test this with `,`. It's used to separate parameters. If you get more that 1 parameter then you need to join parameteres with `,`.

File: src/validators.js [hunk 2]
```diff
@@ -95,3 +95,9 @@ export function validateAlpha(attribute, value) {
 export function validateAlphaNum(attribute, value) {
   return (typeof value === 'string' || typeof value === 'number') && is.alphaNumeric(value)
 }
+
+export function validateRegex(attribute, value, parameters) {
```
Comment (Idered) [inline src/validators.js:99]: You can follow logic implemented here https://github.com/laravel/framework/blob/5.4/src/Illuminate/Validation/Concerns/ValidatesAttributes.php#L1042

File: src/messages.js [hunk 3]
```diff
@@ -20,5 +20,6 @@ export default {
   integer: 'The :attribute must be an integer.',
   accepted: 'The :attribute must be accepted.',
   alpha: 'The :attribute may only contain letters.',
-  alpha_num: 'The :attribute may only contain letters and numbers.'
+  alpha_num: 'The :attribute may only contain letters and numbers.',
+  regex: 'The :attribute must match to the format.'
```
Comment (Idered) [inline src/messages.js]: You can copy messages from here: https://github.com/laravel/laravel/blob/master/resources/lang/en/validation.php

File: tests/index.js [hunk 4]
```diff
@@ -697,4 +697,35 @@ describe('rule', () => {
       return validate(data, rules).then(check).catch(check)
     })
   })
+
+  /*
+   * Regular Expression
+   * ----------------------------------------------------- */
+  describe('#regular_expression', () => {
```
Comment (Idered) [inline tests/index.js:704]: Tests looks good. For future referece, you can check https://github.com/laravel/framework/blob/5.4/tests/Validation/ValidationValidatorTest.php#L2264

File: src/validators.js [hunk 5]
```diff
@@ -95,3 +95,9 @@ export function validateAlpha(attribute, value) {
...
+  const regex = parameters[0] ? new RegExp(`${parameters[0]}`) : undefined
```
Comment (Idered) [inline src/validators.js]: This rule should require just 1 parameter:

```
this.requireParameterCount(1, parameters, 'regex')
```

File: README.md (hunk 1 above)
Comment (Flixow) [reply README.md]: But there is only 1 parameter

Comment (Idered) [reply README.md]: But you can use `,` in regex for example to match date `10th January, 2015`. If you use `,` in regex then you might end up with 2 parameters

File: src/validators.js [hunk 6]
```diff
@@ -95,3 +95,14 @@ export function validateAlpha(attribute, value) {
...
   return (typeof value === 'string' || typeof value === 'number') && is.alphaNumeric(value)
 }
+
+export function validateRegex(attribute, value, parameters) {
+  this.requireParameterCount(1, parameters, 'regex')
+
+  if (is.not.string(value) && is.not.number(value)) {
+    return false
+  }
+  const regex = parameters[0] ? new RegExp(`${parameters[0]}`) : undefined
```
Comment (Idered) [inline src/validators.js]: `this.requireParameterCount` will throw error if parameters[0] doesn't exit so this check is useless.

```
const regex = new RegExp(`${parameters[0]}`)
```
should be enough

**Pull request merged**